core/src/test/java/org/elasticsearch/index/seqno/GlobalCheckpointSyncActionTests.java

```diff
@@ -93,6 +93,19 @@ public void testTranslogSyncAfterGlobalCheckpointSync() throws Exception {
         final Translog translog = mock(Translog.class);
         when(indexShard.getTranslog()).thenReturn(translog);
 
+        final long globalCheckpoint = randomIntBetween(Math.toIntExact(SequenceNumbers.NO_OPS_PERFORMED), Integer.MAX_VALUE);
+        final long lastSyncedGlobalCheckpoint;
+        if (randomBoolean() && globalCheckpoint != SequenceNumbers.NO_OPS_PERFORMED) {
+            lastSyncedGlobalCheckpoint =
+                    randomIntBetween(Math.toIntExact(SequenceNumbers.NO_OPS_PERFORMED), Math.toIntExact(globalCheckpoint) - 1);
+            assert lastSyncedGlobalCheckpoint < globalCheckpoint;
+        } else {
+            lastSyncedGlobalCheckpoint = globalCheckpoint;
```

Comment:
maybe make this `lastSyncedGlobalCheckpoint = globalCheckpoint + randomIntBetween(0, 10);`?

Reply:
That's not realistic? The last synced global checkpoint being greater than the current global checkpoint?